docker/mining.Dockerfile

```diff
@@ -0,0 +1,196 @@
+# We are using five stages:
```

Comment:
Having two different docker files to build Zebra means that we will need to remember to make all our changes in both. But in the past, we have forgotten to make these kinds of duplicate changes.

Can you please implement the mining-specific changes in the original `Dockerfile` using a Docker `ARG` instead?
https://github.com/ZcashFoundation/zebra/issues/6200#issuecomment-1439483686

Gustavo can help with how Docker works, and I can help with the shell scripting.

You might want something like this:
```
ARG ENABLE_MINING_RPCS
...
RUN cargo chef cook --release --features sentry${ENABLE_MINING_RPCS:+,getblocktemplate-rpcs} --recipe-path recipe.json 
```

Reply:
I wasn't able to see this PR previously, but I do agree we need to make it on a single Dockerfile. @arya2 let me know if you'd like some help with this.

Reply:
The new `mining.Dockerfile` was removed and ARGs were added in [`ced6cbe`](https://github.com/ZcashFoundation/zebra/pull/6228/commits/ced6cbeb8317592d236fb890ab4de74868143aec).